common/hash.c

```diff
@@ -41,7 +41,9 @@ DECLARE_GLOBAL_DATA_PTR;
 
 static void reloc_update(void);
 
-#if defined(CONFIG_SHA1) && !defined(CONFIG_SHA_PROG_HW_ACCEL)
+#if ((defined(CONFIG_SHA1) && !defined(CONFIG_SPL_BUILD)) || \
```

Comment:
please use `CONFIG_IS_ENABLED` macro instead, which does that implicitly.
Take a look at `./include/linux/kconfig.h`:

```
 82 /*                                                                              
 83  * CONFIG_IS_ENABLED(FOO) evaluates to                                          
 84  *  1 if CONFIG_SPL_BUILD is undefined and CONFIG_FOO is set to 'y' or 'm',     
 85  *  1 if CONFIG_SPL_BUILD is defined and CONFIG_SPL_FOO is set to 'y' or 'm',   
 86  *  1 if CONFIG_TPL_BUILD is defined and CONFIG_TPL_FOO is set to 'y' or 'm',   
 87  *  0 otherwise.                                                                
 88  */                                                                             
 89 #define CONFIG_IS_ENABLED(option) \                                             
 90         (config_enabled(CONFIG_VAL(option)) ||          \                       
 91          config_enabled(CONFIG_VAL(option##_MODULE)))     
```

Reply:
This file is also used in host tools (USE_HOSTCC mode). In this mode, CONFIG_IS_ENABLED() is unavailable :(

Reply:
Ah, ok, got it.
Then lets keep it as it is. LGTM